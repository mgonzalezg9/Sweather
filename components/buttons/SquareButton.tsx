import React from "react";
import { StyleSheet, TouchableOpacity, View } from "react-native";
import { useThemeColor } from "../Themed";

type SquareButtonType = {
  icon?: JSX.Element;
  onClick: () => void;
};

const SquareButton = ({ icon, onClick, ...props }: SquareButtonType) => {
  const backgroundColor = useThemeColor({}, "tint");

  return (
    <TouchableOpacity
      style={[styles.outline, { backgroundColor }]}
      onPress={onClick}
      {...props}
    >
      <View style={styles.button}>{icon}</View>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  outline: {
    padding: 5,
    width: 50,
    height: 50,
    borderRadius: 10,
  },
  button: {
    display: "flex",
    justifyContent: "center",
    alignItems: "center",
    height: "100%",
    padding: 5,
  },
});

export default SquareButton;
