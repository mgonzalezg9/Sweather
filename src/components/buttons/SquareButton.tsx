import React from "react";
import { Pressable, StyleSheet, View } from "react-native";
import { useThemeColor } from "../Themed";

type SquareButtonType = {
  icon?: JSX.Element;
  onClick: () => void;
};

const SquareButton = ({ icon, onClick, ...props }: SquareButtonType) => {
  const backgroundColor = useThemeColor({}, "tint");

  return (
    <Pressable
      style={[styles.outline, { backgroundColor }]}
      onPress={onClick}
      {...props}
    >
      <View style={styles.button}>{icon}</View>
    </Pressable>
  );
};

const styles = StyleSheet.create({
  outline: {
    padding: 5,
    width: 50,
    height: 50,
    borderRadius: 10,
  },
  button: {
    display: "flex",
    justifyContent: "center",
    alignItems: "center",
    height: "100%",
  },
});

export default SquareButton;
