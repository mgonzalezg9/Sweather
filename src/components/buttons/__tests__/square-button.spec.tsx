import React from "react";
import { Text } from "react-native";
import renderer from "react-test-renderer";
import SquareButton from "../SquareButton";

describe(`SquareButton`, () => {
  it(`renders correctly`, () => {
    const helloWorld = "Hello World";
    const tree = renderer.create(
      <SquareButton onClick={() => {}}>
        <Text>{helloWorld}</Text>
      </SquareButton>
    );

    expect(tree).toMatchSnapshot();
  });
});
