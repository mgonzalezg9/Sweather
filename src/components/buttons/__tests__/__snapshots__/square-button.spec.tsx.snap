// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SquareButton renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "borderRadius": 10,
        "height": 50,
        "padding": 5,
        "width": 50,
      },
      {
        "backgroundColor": "#EC7357",
      },
    ]
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "display": "flex",
        "height": "100%",
        "justifyContent": "center",
        "padding": 5,
      }
    }
  >
    <Text>
      Hello World
    </Text>
  </View>
</View>
`;
